### ISSUE TITLE
Pathfinding + spawning L&S creatures + unload chunks == boom!

### ISSUE THREAD

**Issue by Cervator** [MovingBlocks/Terasology#1304]
Found while testing world loading/unloading. 

* Start a Light & Shadow world
* Place a spawner block or two to get some L&S creatures
* Fly far away so those chunks unload
* Mark an area for one of the pathfinding orders
* Wait a sec - crash! :D

First time I was just flying around furiously (with hspeed) and the stacktrace included a permission error. Second time I just tested exactly like above and provoked a similar error without permission error

First:

```java
23:29:55.622 [Chunk-Generator-3] WARN  o.t.cities.raster.ChunkBrush - Y value of 4 not in range [192..255]
23:30:05.185 [main] INFO  org.terasology.work.WorkBoard - Pathfinding:walkToBlock: open=0 requestable=0 targets=0 
23:30:05.185 [main] INFO  org.terasology.work.WorkBoard - Pathfinding:removeBlock: open=0 requestable=0 targets=0 
23:30:05.185 [main] INFO  org.terasology.work.WorkBoard - Pathfinding:buildBlock: open=0 requestable=0 targets=0 
23:30:10.197 [main] INFO  org.terasology.work.WorkBoard - Pathfinding:walkToBlock: open=0 requestable=0 targets=0 
23:30:10.198 [main] INFO  org.terasology.work.WorkBoard - Pathfinding:removeBlock: open=0 requestable=0 targets=0 
23:30:10.198 [main] INFO  org.terasology.work.WorkBoard - Pathfinding:buildBlock: open=0 requestable=0 targets=0 
23:30:12.428 [Workboard-0] ERROR o.t.u.concurrency.TaskProcessor - Error in thread Workboard-0
java.lang.NullPointerException: null
	at org.terasology.work.WorkBoard$FindWorkTask.run(WorkBoard.java:317) ~[na:na]
	at org.terasology.utilities.concurrency.TaskProcessor.run(TaskProcessor.java:50) ~[na:na]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1145) ~[na:1.7.0_55]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:615) ~[na:1.7.0_55]
	at java.lang.Thread.run(Thread.java:745) ~[na:1.7.0_55]
Unexpected exception
java.security.AccessControlException: access denied ("java.lang.RuntimePermission" "accessClassInPackage.sun.reflect")
	at java.security.AccessControlContext.checkPermission(AccessControlContext.java:372)
	at java.security.AccessController.checkPermission(AccessController.java:559)
	at java.lang.SecurityManager.checkPermission(SecurityManager.java:549)
	at org.terasology.module.sandbox.ModuleSecurityManager.checkPermission(ModuleSecurityManager.java:389)
	at java.lang.SecurityManager.checkPackageAccess(SecurityManager.java:1525)
	at sun.misc.Launcher$AppClassLoader.loadClass(Launcher.java:305)
	at java.lang.ClassLoader.loadClass(ClassLoader.java:358)
	at ch.qos.logback.classic.spi.PackagingDataCalculator.<clinit>(PackagingDataCalculator.java:43)
	at ch.qos.logback.classic.spi.ThrowableProxy.getPackagingDataCalculator(ThrowableProxy.java:135)
	at ch.qos.logback.classic.spi.ThrowableProxy.calculatePackagingData(ThrowableProxy.java:144)
	at ch.qos.logback.classic.spi.LoggingEvent.<init>(LoggingEvent.java:124)
	at ch.qos.logback.classic.Logger.buildLoggingEventAndAppend(Logger.java:440)
	at ch.qos.logback.classic.Logger.filterAndLog_2(Logger.java:433)
	at ch.qos.logback.classic.Logger.error(Logger.java:551)
	at org.terasology.utilities.concurrency.TaskProcessor.run(TaskProcessor.java:60)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1145)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:615)
	at java.lang.Thread.run(Thread.java:745)
23:30:12.560 [main] ERROR o.terasology.engine.TerasologyEngine - Uncaught exception, attempting clean game shutdown
java.lang.NullPointerException: null
	at org.terasology.navgraph.NavGraphSystem.getBlock(NavGraphSystem.java:115) ~[na:na]
	at org.terasology.pathfinding.componentSystem.PathfinderSystem.getBlock(PathfinderSystem.java:103) ~[na:na]
	at org.terasology.minion.move.MinionMoveSystem.setupEntity(MinionMoveSystem.java:80) ~[na:na]
	at org.terasology.minion.move.MinionMoveSystem.worldChanged(MinionMoveSystem.java:48) ~[na:na]
	at org.terasology.minion.move.MinionMoveSystemMethodAccess.invoke(Unknown Source) ~[na:na]
	at org.terasology.entitySystem.event.internal.EventSystemImpl$ByteCodeEventHandlerInfo.invoke(EventSystemImpl.java:497) ~[classes/:na]
	at org.terasology.entitySystem.event.internal.EventSystemImpl.sendStandardEvent(EventSystemImpl.java:268) ~[classes/:na]
	at org.terasology.entitySystem.event.internal.EventSystemImpl.send(EventSystemImpl.java:259) ~[classes/:na]
	at org.terasology.entitySystem.entity.internal.BaseEntityRef.send(BaseEntityRef.java:166) ~[classes/:na]
	at org.terasology.navgraph.NavGraphSystem.update(NavGraphSystem.java:82) ~[na:na]
	at org.terasology.engine.modes.StateIngame.update(StateIngame.java:154) ~[classes/:na]
	at org.terasology.engine.TerasologyEngine.mainLoop(TerasologyEngine.java:466) ~[classes/:na]
	at org.terasology.engine.TerasologyEngine.run(TerasologyEngine.java:268) ~[classes/:na]
	at org.terasology.engine.Terasology.main(Terasology.java:115) ~[classes/:na]
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method) ~[na:1.7.0_55]
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:57) ~[na:1.7.0_55]
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[na:1.7.0_55]
	at java.lang.reflect.Method.invoke(Method.java:606) ~[na:1.7.0_55]
	at com.intellij.rt.execution.application.AppMain.main(AppMain.java:134) ~[idea_rt.jar:na]
23:30:12.560 [main] INFO  o.terasology.engine.TerasologyEngine - Shutting down Terasology...
23:30:12.573 [main] INFO  o.t.n.internal.NetworkSystemImpl - Client disconnected: Player
23:30:12.573 [main] INFO  o.t.l.console.internal.ConsoleImpl - [NOTIFICATION] Player "Player" has left the game
23:30:12.580 [main] INFO  o.t.n.internal.NetworkSystemImpl - Network shutdown
23:30:13.484 [main] WARN  org.reflections.Reflections - given scan urls are empty. set urls in the configuration

Process finished with exit code 0
```

Second:

```java
23:32:43.919 [Chunk-Generator-0] WARN  o.t.cities.raster.ChunkBrush - Y value of 21 not in range [192..255]
23:32:43.956 [main] ERROR o.terasology.engine.TerasologyEngine - Uncaught exception, attempting clean game shutdown
java.lang.NullPointerException: null
	at org.terasology.navgraph.NavGraphSystem.getBlock(NavGraphSystem.java:115) ~[na:na]
	at org.terasology.pathfinding.componentSystem.PathfinderSystem.getBlock(PathfinderSystem.java:103) ~[na:na]
	at org.terasology.minion.move.MinionMoveSystem.setupEntity(MinionMoveSystem.java:80) ~[na:na]
	at org.terasology.minion.move.MinionMoveSystem.worldChanged(MinionMoveSystem.java:48) ~[na:na]
	at org.terasology.minion.move.MinionMoveSystemMethodAccess.invoke(Unknown Source) ~[na:na]
	at org.terasology.entitySystem.event.internal.EventSystemImpl$ByteCodeEventHandlerInfo.invoke(EventSystemImpl.java:497) ~[classes/:na]
	at org.terasology.entitySystem.event.internal.EventSystemImpl.sendStandardEvent(EventSystemImpl.java:268) ~[classes/:na]
	at org.terasology.entitySystem.event.internal.EventSystemImpl.send(EventSystemImpl.java:259) ~[classes/:na]
	at org.terasology.entitySystem.entity.internal.BaseEntityRef.send(BaseEntityRef.java:166) ~[classes/:na]
	at org.terasology.navgraph.NavGraphSystem.update(NavGraphSystem.java:82) ~[na:na]
	at org.terasology.engine.modes.StateIngame.update(StateIngame.java:154) ~[classes/:na]
	at org.terasology.engine.TerasologyEngine.mainLoop(TerasologyEngine.java:466) ~[classes/:na]
	at org.terasology.engine.TerasologyEngine.run(TerasologyEngine.java:268) ~[classes/:na]
	at org.terasology.engine.Terasology.main(Terasology.java:115) ~[classes/:na]
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method) ~[na:1.7.0_55]
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:57) ~[na:1.7.0_55]
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[na:1.7.0_55]
	at java.lang.reflect.Method.invoke(Method.java:606) ~[na:1.7.0_55]
	at com.intellij.rt.execution.application.AppMain.main(AppMain.java:134) ~[idea_rt.jar:na]
23:32:43.956 [main] INFO  o.terasology.engine.TerasologyEngine - Shutting down Terasology...
23:32:43.969 [main] INFO  o.t.n.internal.NetworkSystemImpl - Client disconnected: Player
23:32:43.970 [main] INFO  o.t.l.console.internal.ConsoleImpl - [NOTIFICATION] Player "Player" has left the game
23:32:43.981 [main] INFO  o.t.n.internal.NetworkSystemImpl - Network shutdown
23:32:44.734 [main] WARN  org.reflections.Reflections - given scan urls are empty. set urls in the configuration

Process finished with exit code 0
```

**Comment by Cervator** [comment]
@flo - I'm thinking this is related to your comments in #1503? Since it involves stuff unloading or not at range

**Comment by Cervator** [comment]
Closing as fixed by #2199 and other recent BT/L&S changes. Not that the L&S mobs suddenly work fully, but this crash doesn't occur anymore.
